src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.cs

```diff
@@ -67,12 +66,25 @@ private GCHandle(object? value, GCHandleType type)
         /// <returns>A new GC handle that protects the object.</returns>
         public static GCHandle Alloc(object? value, GCHandleType type) => new GCHandle(value, type);
 
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            IntPtr handle = _handle;
```

Comment:
❔ Can we not `Interlocked.Exchange` here?

```suggestion
            IntPtr handle = Interlocked.Exchange(ref _handle, IntPtr.Zero);
```

Reply:
https://github.com/dotnet/runtime/issues/54792#issuecomment-870684409

Reply:
For context: I've removed `Interlocked.Exchange` following the conversation between Jan and Stephen [here](https://github.com/dotnet/runtime/issues/54792#issuecomment-870684409) 🙂